tests/HttpServerTest.php

```diff
@@ -706,4 +707,59 @@ public function testSendChunkedEncodedRequestReceiveResponseAndTcpConnectionCant
 
         $this->connection->emit('data', array("GET / HTTP/1.1\r\nTransfer-Encoding: chunked\r\n\r\n3\r\nbla\r\n0\r\n\r\n"));
     }
+
+    public function testServerRequestWillBeUsedForCallable()
+    {
+        $callback = function (RequestInterface $request) {
+            if (! method_exists($request, 'getCookieParams')) {
```

Comment:
what is this supposed to do? also, unreachable `return` after `throw`? Should this use a mock that asserts the right instance instead?